amm/contracts/stable_pool/lib.rs

```diff
@@ -0,0 +1,1033 @@
+    //             vec![AccountId::from([1u8; 32]), AccountId::from([2u8; 32])],
+    //             vec![6, 12],
+    //             1,
+    //             AccountId::from([0u8; 32]),
+    //             None,
+    //         )
+    //         .map_err(|err| panic!("Contract instantiation error: {err:?}"))
+    //         .unwrap();
+    //         let amount: u128 = 1_000_000_000_000; // 1000000.000000
+    //         let expect_amount: u128 = amount * 10u128.pow((TOKEN_TARGET_DECIMALS - 6) as u32); // 1000000.000000000000
+    //         assert_eq!(
+    //             stable_pool_contract.amount_to_r_amount(amount, 0),
+    //             Ok(expect_amount)
+    //         );
+    //         assert_eq!(
+    //             stable_pool_contract
+    //                 .r_amount_to_amount(expect_amount, 0)
+    //                 .unwrap(),
+    //             amount
+    //         );
+    //         let amount: u128 = 1_000_000_000_000_000_000; // 1000000.000000
+    //         let expect_amount: u128 = amount * 10u128.pow((TOKEN_TARGET_DECIMALS - 12) as u32);
+    //         assert_eq!(
+    //             stable_pool_contract.amount_to_r_amount(amount, 1),
+    //             Ok(expect_amount)
+    //         );
+    //         assert_eq!(
+    //             stable_pool_contract
+    //                 .r_amount_to_amount(expect_amount, 1)
+    //                 .unwrap(),
+    //             amount
+    //         );
+    //     }
+
+    //     #[test]
+    //     fn amount_to_comparable_and_back_2() {
+    //         let stable_pool_contract = StablePoolContract::new_stable(
+    //             vec![AccountId::from([1u8; 32]), AccountId::from([2u8; 32])],
+    //             vec![0, 18],
+    //             1,
+    //             AccountId::from([0u8; 32]),
+    //             Some(AccountId::from([0u8; 32])),
+    //         )
+    //         .map_err(|err| panic!("Contract instantiation error: {err:?}"))
+    //         .unwrap();
+    //         let amount: u128 = 1_000_000; // 1000000
+    //         let expect_amount: u128 = amount * 10u128.pow(RATE_DECIMALS as u32); // 1000000.000000000000000000
+    //         assert_eq!(
+    //             stable_pool_contract.amount_to_r_amount(amount, 0),
+    //             Ok(expect_amount)
+    //         );
+    //         assert_eq!(
+    //             stable_pool_contract.r_amount_to_amount(expect_amount, 0),
+    //             Ok(amount)
+    //         );
+    //         let amount: u128 = 1_000_000_000_000_000_000_000_000_000_000; // 1000000.000000000000000000
+    //         let expect_amount: u128 = amount;
+    //         assert_eq!(
+    //             stable_pool_contract.amount_to_r_amount(amount, 1),
+    //             Ok(expect_amount)
+    //         );
+    //         assert_eq!(
+    //             stable_pool_contract.r_amount_to_amount(expect_amount, 1),
+    //             Ok(amount)
+    //         );
+    //     }
+    // }
```

Comment:
What about these tests? Are we working on bringing them back or do we remove them entirely?

Reply:
Yeah, some tests need to be added here. The old tests stopped making sense because they were mostly about decimal calculations which changed completely. Removed the commented tests in the PR